modules/widgets/social-icons.php

```diff
@@ -137,11 +137,27 @@ public function widget( $args, $instance ) {
 									$found_icon = false;
 
 								foreach ( $social_icons as $social_icon ) {
-									if ( false !== stripos( $icon['url'], $social_icon['url'] ) ) {
-										echo '<span class="screen-reader-text">' . esc_attr( $social_icon['label'] ) . '</span>';
-										echo $this->get_svg_icon( array( 'icon' => esc_attr( $social_icon['icon'] ) ) );
-										$found_icon = true;
-										break;
+									
+									// Check if the $social_icon['url'] is an array ('Feed' url is provided as an array of url components)
+									if ( is_array( $social_icon['url'] ) ) {
```

Comment:
That's a good approach. Would it be possible to simplify things even more by always having an array, but having an array with a single value in those cases? This way we would simplify things here.

Reply:
Ah yep, that would be definitely be a good way to simplify it and avoid the repetition!

Reply:
Ok, I've made that change now. I realised as I was committing that change that this means that the ten different 'amazon.xx' entries (amazon.cn, amazon.in, amazon.fr etc) in the `$social_links_icons` array could now be reduced to one with all the domain variations in that single entry.

Reply:
> this means that the ten different 'amazon.xx' entries (amazon.cn, amazon.in, amazon.fr etc) in the $social_links_icons array could now be reduced to one with all the domain variations in that single entry.

Let's do it! :)

Reply:
Ok, done. I've committed a change that reduces Amazon, Google, Discord and WordPress domains into single entries instead.